### DISCUSSION TITLE
[Authorization service] Permissions proxy mechanism

### DISCUSSION THREAD

**Discussion by FranL27** [mia-platform/community#159] [Q&A]
Hi all, 
For a client we are adopting a multi-project architecture where requests coming to PROJECT-A (exposed on the Internet) are authorized and proxied towards PROJECT-B. PROJECT-B trusts the authorization applied by PROJECT-A and therefore does not apply any authorization logic. 
Now we want to apply a finer-grained authorization logic and therefore, for some endpoints exposed on PROJECT-A, we apply no authorization logic and leave this responsibility to PROJECT-B. The authorization logic we want to apply is based on permissions, but these are only retrieved at the level of PROJECT-A and are not proxied to PROJECT-B unlike what happens for groups that are proxied correctly. 
It seems that permissions are only evaluated on the project where the call lands and are no longer proxied. Is there any way to get the permissions proxied to PROJECT-B as well?
Attached are the logs of the authorization calls made by authorization-service on PROJECT-A and PROJECT-B respectively.
Thank you. 

Authorization-service on PROJECT-A
![Screenshot from 2023-01-05 14-58-34](https://user-images.githubusercontent.com/29659126/210801487-f5c8505a-6e27-4986-ac91-abed5ef94ec6.png)

Authorization-service on PROJECT-B
![Screenshot from 2023-01-05 14-58-46](https://user-images.githubusercontent.com/29659126/210801492-e154b1c3-1af5-405c-bc91-5a35550eadff.png)

**Comment by davidebianchi** [comment]
Where are the permissions? Do you have a JWT with the permissions set? In this case, you should retrieve from the JWT the permissions.
Otherwise, permissions could be set inside a specific header to proxy (set by env var `USER_PROPERTIES_HEADER_KEY `), maybe it is possible to make something in this way.

**Reply by FranL27** [reply]
The permissions are inside the JWT. How do I specify in the authorization-service to retrieve them from the JWT?
